text/0000-add-scratch-stack.md

```diff
@@ -0,0 +1,161 @@
+# Create scratch stack
+
+## Summary
+
+The Paketo stacks maintainers should create and maintain a scratch stack that
+can run statically-linked applications (e.g. golang, Rust). The Builder team does not
+need to create a builder for this image, but may do so if they wish.
+
+Additionally, the golang buildpack should gain support for this new scratch
+stack, both in order to provide a useful experience to golang application
+developers as well as to enable integration tests for this new stack.
+
+## Motivation
+
+Some applications consist of a single
+[statically-linked](https://en.wikipedia.org/wiki/Static_library) executable.
+This is most common in golang, but is also often utilized in rust. It is also
+common in languages/toolchains not supported by Paketo like C/C++. These
+statically-linked binaries can be run on a minimally-modified empty container
+image (AKA "scratch" image) with few-to-no packages/libraries. Scratch images
+are a common choice of container image for statically-linked executables as it
+minimizes the size (in MB) of the image as well as the number of installed
+packages - in turn reducing the surface area for security issues.
+
+Currently, in order to build these statically-linked applications using Paketo
+buildpacks (and aiming for the smallest possible resultant run image) users
+must either opt for the [tiny
+stack](https://github.com/paketo-buildpacks/jammy-tiny-stack) or build their
+own scratch stack. We have evidence of users doing both of these things.
+Neither are satisfactory to users, however, as the tiny stack contains more
+packages than is necessary for those applications, and users would prefer not
+to maintain their own stacks if Paketo is willing and able to do so.
+
+Perhaps most importantly, we are [considering adding more packages to the tiny
+stack](https://github.com/paketo-buildpacks/rfcs/pull/231/files), and the
+biggest objection to that proposal is that it will further increase the size
+and surface area of the tiny image for statically-linked binaries. This could,
+in-turn, make Paketo less attractive for developers of these applications.
+
+Creating a scratch stack will decouple these use cases, allowing us to tailor
+the tiny stack to applications which want more packages, while also providing a
+better experience for statically-linked binaries.
+
+## Detailed Explanation
+
+### Repository/Image Location
+
+We will create a new Github repository at
+`paketo-buildpacks/jammy-scratch-stack` and a new pair of images on Dockerhub
+at `paketobuildpacks/build-jammy-scratch` and
+`paketobuildpacks/run-jammy-scratch`.
+
+We will adopt the tag system used for Jammy - specifically semver (i.e.
+'X.Y.Z') and 'latest' tags.
+
+We will not create a builder repository or image, though we reserve the option
+for the Builder team to do so in the future if that is requested and approved.
```

Comment:
I'm kind of wondering why no builder will be created? Users of the scratch stack will then have to create their own builder. To me, if we're going to support this we might as well just support it end-to-end.

Reply:
I mostly didn't want to sign the builder team up for work in this RFC. I agree that it is easier to create a scratch Builder with the same process we currently use for other builders. I could see value in at least a buildpackless builder.

@paketo-buildpacks/builders-maintainers do you think the burden of creating a buildpackless builder for the proposed scratch image is acceptable and something you would be willing to take on?

Reply:
Given that we'd only be adding one builder (Jammy, with no buildpacks), I'd be open to the addition. The maintenance burden here would be extremely low and it'd smooth the onboarding process for static-stack users. @paketo-buildpacks/builders-maintainers any other opinions on this?

Reply:
I'll update the text to say that the builder team will make a buildpackless builder. @paketo-buildpacks/builders-maintainers can still voice any objections to the new text :)